app/core/settings.py

```
@@ -251,5 +257,51 @@ def get_settings() -> Settings:
+        
+        # Validate Gemini API key format (at least 20 allowed chars, starts with allowed prefix)
+        if settings.gemini_api_key:
+            key = settings.gemini_api_key.strip()
+            gemini_key_pattern = re.compile(r"^[A-Za-z0-9_\-]{20,}$")
+            if not gemini_key_pattern.match(key):
+                raise ValueError("GEMINI_API_KEY appears to be invalid (incorrect format)")
+        
+        # Validate rate limiting configuration
+        try:
+            from app.core.rate_limiting.config import RateLimitConfig
+            config = RateLimitConfig.from_environment()
```

Ruff (0.12.2)

Local variable `config` is assigned to but never used

Remove assignment to unused variable `config`

(F841)